scripts/makebumpver.j2

```diff
@@ -0,0 +1,467 @@
+
+        # gather all unprocessed command line arguments
+        parser.add_argument(nargs=argparse.REMAINDER, dest="unknown_arguments")
+
+        self.args = parser.parse_args()
+        if self.args.debug:
+            logging.basicConfig(level=logging.DEBUG)
+
+        if self.args.bump_major_version and self.args.add_version_number:
+            print("The --bump-major-version and --add-version-number options can't be used at the same time.",
+                  file=sys.stderr)
+            sys.exit(1)
+
+        if self.args.unknown_arguments:
+            parser.print_usage()
+            args_string = " ".join(self.args.unknown_arguments)
+            print("unknown arguments: %s" % args_string, file=sys.stderr)
+            sys.exit(1)
+
+        if not os.path.isfile(self.configure) and not os.path.isfile(self.spec):
+            print("You must be at the top level of the anaconda source tree.", file=sys.stderr)
+            sys.exit(1)
+
+        # general initialization
+        log.debug("%s", self.args)
+        self._jira = JIRAValidator()
+
+        self.gituser = self._git_config('user.name')
+        self.gitemail = self._git_config('user.email')
+
+        self.name = self.args.name
+        self.version = self.args.version
+        self.release = self.args.release
+        self.new_release = self.args.new_release or self.release
+        self.bugreport = self.args.bugreporting_email
+        self.ignore = self.args.ignored_commits
+        self.skip = self.args.skip_all_checks
+        self.dry_run = self.args.dry_run
+
+        self.git_branch = None
+
+        # RHEL release number or None (also fills in self.git_branch)
+        self.rhel = self._is_RHEL()
+
+    def _git_config(self, field):
+        proc = run_program(['git', 'config', field])
+        return proc[0].strip('\n')
+
+    def _increment_version(self, bump_major_version=False, add_version_number=False):
+        fields = self.version.split('.')
+        if add_version_number:  # add another version number to the end
+            fields.append(str(int(DEFAULT_ADDED_VERSION_NUMBER)))
+        elif bump_major_version:  # increment major version
+            fields[0] = str(int(fields[0]) + VERSION_NUMBER_INCREMENT)  # bump major version
+            fields[1] = str(int(DEFAULT_MINOR_VERSION_NUMBER))  # reset minor version to 1
+        else:  # minor version bump
+            fields[-1] = str(int(fields[-1]) + VERSION_NUMBER_INCREMENT)
+        new = ".".join(fields)
+        return new
+
+    def _is_RHEL(self):
+        proc = run_program(['git', 'branch'])
+        lines = [x for x in proc[0].strip('\n').split('\n') if x.startswith('*')]
+
+        if lines == [] or len(lines) > 1:
+            return False
+
+        fields = lines[0].split(' ')
+
+        if len(fields) == 2:
+            self.git_branch = fields[1]
+        else:
+            return False
+
+        if self.git_branch.startswith('rhel'):
+            branch_pattern = r"^rhel(\d+)-(.*)"
```

Comment:
Is this branch pattern still used?

Reply:
I think it is not. Maybe I'd remove it in a follow-up separate PR, but I'd be fine with a commit in this set as well.